libevmasm/PeepholeOptimiser.cpp

```diff
@@ -146,6 +146,35 @@ struct OpStop: SimplePeepholeOptimizerMethod<OpStop>
 	}
 };
 
+struct OpReturnRevert: SimplePeepholeOptimizerMethod<OpReturnRevert>
+{
+	static bool applySimple(
+		AssemblyItem const& _op,
+		AssemblyItem const& _push,
+		AssemblyItem const& _pushOrDup,
+		AssemblyItem const& _returnRevert,
+		std::back_insert_iterator<AssemblyItems> _out
+	)
+	{
+		if (
```

Comment:
This mechanism could even be extended more:
We have op1, op2, op3, returnRevert where:
 - op1 is side-effect free
 - op2.args == 0, op2.return >= 1
 - op3.args <= op2.return
 - (op2.return + op3.return) - (op2.args + op3.args) >= 2
 - returnRevert is return or revert